distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -840,58 +841,119 @@ Text.drop_last count =
+                iterator = BreakIterator.getCharacterInstance
+                iterator.setText text
 
-   ! What is a Character?
-     A character is defined as an Extended Grapheme Cluster, see Unicode
-     Standard Annex 29. This is the smallest unit that still has semantic
-     meaning in most text-processing applications.
+                loop index start_char end_char start_index =
+                    if index == end then (Range start_index (start_char)) else
+                        @Tail_Call loop (index + 1) end_char iterator.next (if index == start then start_char else start_index)
 
-   > Example
-     Make a new text from the first two characters of "boo".
+                loop 0 0 iterator.next Nothing
 
-         "boo".take_first 2
-Text.take_first : Integer -> Text
-Text.take_first count =
-    iterator = BreakIterator.getCharacterInstance
-    iterator.setText this
-    iterator.first
-    boundary = iterator.next count
-    if boundary == -1 then this else Text_Utils.substring this 0 boundary
+## ALIAS first, last, left, right, mid, substring
+   Creates a new Text by selecting the specified range of the input.
 
-## Creates a new text by selecting the last `count` characters of `this`,
-   returning `this` if `count` is greater than or equal to the length of `this`.
+   This can select a section of text from the beginning, end, or middle of the
+   input using various criteria defined by the range parameter.
 
    Arguments:
-   - count: The number of characters to take from the end of `this`.
+   - range: The section of the this text to return.
+     If a `Text_Sub_Range`, then the selection is interpreted following the rules of that type.
+     If a `Range`, the selection is specified by two indices, from and to.
+
+   Returns:
+   The part of the input as specified by the range parameter.
+
+   > Examples
+     Various different ways to take part of "Hello World!"
+
+         "Hello World!".take First.new == "H"
+         "Hello World!".take (First 5) == "Hello"
+         "Hello World!".take (First 0) == ""
+         "Hello World!".take Last.new == "!"
+         "Hello World!".take (Last 6) == "World!"
+         "Hello World!".take (Before " ") == "Hello"
+         "Hello World!".take (Before_Last "o") == "Hello W"
+         "Hello World!".take (After " ") == "World!"
+         "Hello World!".take (After_Last "o") == "rld!"
+         "Hello World!".take (While c->c!=" ") == "Hello"
+         "Hello World!".take (Range 3 5) == "lo"
+         "Hello World!".take (Range -3 -1) == "ld"
+         "Hello World!".take (Range -3 Nothing) == "ld!"
+         "Hello World!".take (Range 5 Nothing) == " World!"
+         "Hello World!".take (Range 5 12) == " World!"
+         "Hello World!".take (Range 12 12) == ""
+Text.take : (Text_Sub_Range | Range) -> Text ! Index_Out_Of_Bounds_Error
+Text.take range =
+    char_range = case range of
+        Range _ _ -> here.range_to_char_indices this range
+        _ -> range.to_char_range this
```

Comment:
```suggestion
        Range _ _ -> here.range_to_char_indices this range
        Text_Sub_Range -> range.to_char_range this
```

would be cool to be able to do it, but I think it may be impossible. Did you try it? If it doesn't work, it may be worth reporting to the engine - if I understand correctly the typeset design, this should be possible - and in this case would make the code much clearer than a `_`. (It would also catch errors earlier, rather than failing with `some random value does not define a to_char_range method`).

Reply:
didn't work when I tried it earlier so ended up with this approach.

will raise with engine team